pkg/driver/driver.go

```diff
@@ -330,6 +330,65 @@ func (driver *Driver) GetVolumeByID(id string, secrets map[string]string) (*mode
 	return volume, nil
 }
 
+// GetVolumeByName retrieves the volume instance by name from the CSP if exists
+func (driver *Driver) GetVolumeByName(name string, secrets map[string]string) (*model.Volume, error) {
+	log.Trace(">>>>> GetVolumeByName, name: ", name)
+	defer log.Trace("<<<<< GetVolumeByName")
+
+	var volume *model.Volume
+	var err error
+	// When secrets specified
+	if len(secrets) != 0 {
+		log.Tracef("Secrets are provided. Checking with this particular storage provider.")
+
+		// Get Storage Provider
+		storageProvider, err := driver.GetStorageProvider(secrets)
+		if err != nil {
+			log.Error("err: ", err.Error())
+			return nil, status.Error(codes.Internal, "Failed to get storage provider from secrets")
+		}
+
+		// check if the volume exists
+		volume, err = storageProvider.GetVolumeByName(name)
+		if err != nil {
+			log.Error("err: ", err.Error())
+			return nil, status.Error(codes.Internal, fmt.Sprintf("Error while attempting to get volume %s", name))
+		}
+	} else {
+		log.Tracef("Secrets not provided. Checking all known storage providers.")
```

Comment:
I'm not a fan of doing this by name.  The ID is a UUID.  The name is not.  We should probably only support the case where the secrets are provided.

Reply:
Agreed. Will change to lookup with specific provider for the given secrets.